openedx/core/djangoapps/catalog/management/commands/cache_programs.py

```diff
@@ -25,18 +31,107 @@ class Command(BaseCommand):
     help = "Rebuild the LMS' cache of program data."
 
     def handle(self, *args, **options):
-        catalog_integration = CatalogIntegration.current()
-        username = catalog_integration.service_username
+        if not waffle.switch_is_active("populate-multitenant-programs"):
+            catalog_integration = CatalogIntegration.current()
+            username = catalog_integration.service_username
 
-        try:
-            user = User.objects.get(username=username)
-            client = create_catalog_api_client(user)
-        except User.DoesNotExist:
-            logger.error(
-                'Failed to create API client. Service user {username} does not exist.'.format(username)
-            )
-            raise
+            try:
+                user = User.objects.get(username=username)
+                client = create_catalog_api_client(user)
+            except User.DoesNotExist:
+                logger.error(
+                    'Failed to create API client. Service user {username} does not exist.'.format(username)
+                )
+                raise
+
+            try:
+                querystring = {
+                    'exclude_utm': 1,
+                    'status': ('active', 'retired'),
+                    'uuids_only': 1,
+                }
+
+                logger.info('Requesting program UUIDs.')
+                uuids = client.programs.get(**querystring)
+            except:  # pylint: disable=bare-except
+                logger.error('Failed to retrieve program UUIDs.')
+                raise
+
+            total = len(uuids)
+            logger.info('Received {total} UUIDs.'.format(total=total))
+
+            programs = {}
+            failure = False
+            for uuid in uuids:
+                try:
+                    logger.info('Requesting details for program {uuid}.'.format(uuid=uuid))
+                    program = client.programs(uuid).get(exclude_utm=1)
+
+                    cache_key = PROGRAM_CACHE_KEY_TPL.format(uuid=uuid)
+                    programs[cache_key] = program
+                except:  # pylint: disable=bare-except
+                    logger.exception('Failed to retrieve details for program {uuid}.'.format(uuid=uuid))
+                    failure = True
+
+                    continue
```

Comment:
This code (everything in if not waffle.switch_is_active("populate-multitenant-programs"): ) is going to go away once we remove the waffle switch.